libraries/eventformatter/impl/src/main/kotlin/io/element/android/libraries/eventformatter/impl/isme/DefaultIsMe.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.eventformatter.impl.isme
+
+import com.squareup.anvil.annotations.ContributesBinding
+import io.element.android.libraries.di.SessionScope
+import io.element.android.libraries.matrix.api.MatrixClient
+import io.element.android.libraries.matrix.api.core.SessionId
+import javax.inject.Inject
+
+@ContributesBinding(SessionScope::class)
+class DefaultIsMe @Inject constructor(
+    private val matrixClient: MatrixClient,
+) : IsMe {
+    override fun invoke(sessionId: SessionId): Boolean {
+        return sessionId == matrixClient.sessionId
```

Comment:
I introduced the `IsMe` interface to avoid having to pass a `FakeMatrixClient` (because of the new TestScope thing). Also I think it will be easier to add some test on the classes which are using it.
I like to isolate functionnality. A MatrixClient can do lots of thing, and here we just need the `SessionId`. I have considered injecting `SessionId` but this is not possible since this is a Kotlin class and  this does not work with Dagger.

Reply:
> I like to isolate functionnality

This is not isolating functionality because there is nothing that prevents other developers from not using `IsMe` and keep using `MatrixClient`.

Codebase newcomers will face more cognitive load because instead of dealing only with `MatrixClient` they will also have to deal with `IsMe` and understand when to use one or the other. Many might not just realize `IsMe` even exists and therefore won't use it.
This will also increase the load for reviewers which might need to spot "wrong" usages of `MatrixClient` and suggest to swap them with `IsMe`.

I agree that it would be nice to have an explicit function to check if a sessionId belongs to the current client's: an extension over `MatrixClient` would achieve this without introducing any complexity:
```
fun MatrixClient.isMe(sessionId: SessionId): Boolean {
    return sessionId == this.sessionId
}
```